core/src/main/scala/filodb.core/reprojector/Reprojector.scala

```diff
@@ -65,14 +65,13 @@ class DefaultReprojector(columnStore: ColumnStore)
       (partition, helper.getSegment(sortKey))
     }.map { case ((partition, (segStart, segUntil)), segmentRowsIt) =>
       // For each segment grouping of rows... set up a Segment
-      val keyRange = KeyRange(dataset.name, partition, segStart, segUntil)
+      val keyRange = KeyRange(dataset.name, partition, segStart, segStart)
       val segment = new RowWriterSegment(keyRange, memTable.projection.columns)
       logger.debug(s"Created new segment $segment for encoding...")
 
       // Group rows into chunk sized bytes and add to segment
-      segmentRowsIt.grouped(dataset.options.chunkSize).foreach { chunkRowsIt =>
-        val chunkRows = chunkRowsIt.toSeq
-        segment.addRowsAsChunk(chunkRows)
+      while (segmentRowsIt.nonEmpty) {
+        segment.addRowsAsChunk(segmentRowsIt.take(dataset.options.chunkSize))
       }
```

Comment:
Maybe a for comp instead?

     for (n <- segmentRowsIt) {
        segment.addRowsAsChunk(segmentRowsIt.take(dataset.options.chunkSize))
     }